packages/evalite/src/ai-sdk.ts

```diff
@@ -83,38 +79,60 @@ const processPromptForTracing = (
   });
 };
 
-export const traceAISDKModel = (model: LanguageModelV1): LanguageModelV1 => {
+export const traceAISDKModel = (model: LanguageModelV2): LanguageModelV2 => {
   if (!shouldReportTrace()) return model;
-  return experimental_wrapLanguageModel({
+  return wrapLanguageModel({
     model,
     middleware: {
       wrapGenerate: async (opts) => {
         const start = performance.now();
         const generated = await opts.doGenerate();
         const end = performance.now();
 
+        const textContent = generated.content
+          .filter((c) => c.type === "text")
+          .map((c) => c.text)
+          .join("");
+
+        const toolCalls = generated.content
+          .filter((c) => c.type === "tool-call")
+          .map((c) =>
+            c.type === "tool-call"
+              ? {
+                  toolName: c.toolName,
+                  input: c.input,
+                  toolCallId: c.toolCallId,
+                }
+              : null
+          )
+          .filter(Boolean);
+
         reportTrace({
           output: {
-            text: generated.text ?? "",
-            toolCalls: generated.toolCalls,
+            text: textContent,
+            toolCalls,
           },
           input: processPromptForTracing(opts.params.prompt),
-          usage: generated.usage,
+          usage: {
+            inputTokens: generated.usage.inputTokens ?? 0,
+            outputTokens: generated.usage.outputTokens ?? 0,
+            totalTokens: generated.usage.totalTokens ?? 0,
+          },
```

Comment:
Not particularly proud of this one but `Evalite.Trace` uses non optional values whereas AI SDK token values may be undefined.